NEWS.md

```diff
@@ -0,0 +1,276 @@
+was evidently confused about the type and expected to have a `Traversal[Method]` in hand that contains the single required `Method`.
+This code used to compile, due to an implicit conversion from a single node to `Traversal[Node]`, and due to `Traversal[Node]` having the
+member function `head`, mixed in from `IterableOps`.
+
+Such code does not compile anymore. The solution is to remove the superfluous `.head` call, i.e.
+```
+val method           = methodReturn.method
+```
+
+Another example happened in the following code:
+```
+  def globalFromLiteral(lit: Literal): Traversal[Expression] = lit
+    .where(_.inAssignment.method.nameExact("<module>", ":package"))
+    .inAssignment
+    .argument(1)
+```
+This code used to work due to the same implicit conversion, `where` being a member method of `Traversal`. This now needs to be explicit, via either
+```
+  def globalFromLiteral(lit: Literal): Traversal[Expression] = Iterator(lit)
+    .where(_.inAssignment.method.nameExact("<module>", ":package"))
+    .inAssignment
+    .argument(1)
+```
+or the fluent
+```
+  def globalFromLiteral(lit: Literal): Traversal[Expression] = lit.start
+    .where(_.inAssignment.method.nameExact("<module>", ":package"))
+    .inAssignment
+    .argument(1)
+```
+
+Another example happened in the following snippets:
+```
+  implicit def toDdgNodeDot(traversal: IterableOnce[Method]): DdgNodeDot =
+    new DdgNodeDot(traversal)
+//...
+class DdgNodeDot(val traversal: Traversal[Method]) extends AnyVal {
+//...
+```
+This code used to work due to an implicit conversion from `IterableOnce` to `Traversal`. We do not have this implicit conversion anymore, and the code should instead read:
+```
+  implicit def toDdgNodeDot(traversal: IterableOnce[Method]): DdgNodeDot =
+    new DdgNodeDot(traversal.iterator)
+```
+This is an important design pattern in our extensions methods: We accept `IterableOnce` for our extensions, in order to avoid users
+having to type `.iterator` everywhere in their code, and we always produce `Iterator`, in order to make it easy to reason about the semantics.
+
+This issue affected 27 files in joern.
+
+#### Use of the `count` step
+```
+[error] joern/semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/NodeSteps.scala:90:54: missing argument list for method count in trait IterableOnceOps
+[error] Unapplied methods are only converted to functions when a function type is expected.
+[error] You can make this conversion explicit by writing `count _` or `count(_)` instead of `count`.
+[error]       .repeat(_.in(edgeType))(_.until(_.in(edgeType).count.filter(_ == 0)))
+```
+There used to be a step `count: Traversal[Int]` that turns a traversal into a new traversal with a single integer element that contains the number of items in
+the original traversal.
+
+Unfortunately there is a nameclash with an existing `count` method on `IterableOnceOps`. The old code had no problem with that, since a direct 
```

Comment:
```suggestion
Unfortunately there is a name clash with an existing `count` method on `IterableOnceOps`. The old code had no problem with that, since a direct 
```